src/Elastic.Apm/Cloud/AwsCloudMetadataProvider.cs

```diff
@@ -0,0 +1,92 @@
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Elastic.Apm.Api;
+using Elastic.Apm.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Elastic.Apm.Cloud
+{
+	/// <summary>
+	/// Provides cloud metadata for Amazon Web Services (AWS)
+	/// </summary>
+	internal class AwsCloudMetadataProvider : ICloudMetadataProvider
+	{
+		internal const string TokenUri = "http://169.254.169.254/latest/api/token";
+		internal const string MetadataUri = "http://169.254.169.254/latest/dynamic/instance-identity/document";
+		internal const string Name = "aws";
+
+		private readonly HttpMessageHandler _handler;
+		private readonly IApmLogger _logger;
+
+		public AwsCloudMetadataProvider(IApmLogger logger) : this(new HttpClientHandler(), logger)
+		{
+		}
+
+		internal AwsCloudMetadataProvider(HttpMessageHandler handler, IApmLogger logger)
+		{
+			_handler = handler;
+			_logger = logger.Scoped(nameof(AwsCloudMetadataProvider));
+		}
+
+		/// <inheritdoc />
+		public string Provider { get; } = Name;
+
+		/// <inheritdoc />
+		public async Task<Api.Cloud> GetMetadataAsync()
+		{
+			var client = new HttpClient(_handler, false);
+			try
+			{
+				string awsToken;
+				using (var requestMessage = new HttpRequestMessage(HttpMethod.Put, TokenUri))
+				{
+					requestMessage.Headers.Add("X-aws-ec2-metadata-token-ttl-seconds", "300");
+					using var tokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(3));
+					var responseMessage = await client.SendAsync(requestMessage, tokenSource.Token).ConfigureAwait(false);
+					awsToken = await responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
+				}
+
+				JObject metadata;
+				using (var requestMessage = new HttpRequestMessage(HttpMethod.Put, MetadataUri))
+				{
+					requestMessage.Headers.Add("X-aws-ec2-metadata-token", awsToken);
+					using var tokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(3));
+					var responseMessage = await client.SendAsync(requestMessage, tokenSource.Token).ConfigureAwait(false);
+
+					using var stream = await responseMessage.Content.ReadAsStreamAsync().ConfigureAwait(false);
+					using var streamReader = new StreamReader(stream, Encoding.UTF8);
+					using var jsonReader = new JsonTextReader(streamReader);
+
+					var serializer = new JsonSerializer();
+					metadata = serializer.Deserialize<JObject>(jsonReader);
+				}
+
+				return new Api.Cloud
+				{
+					Account = new CloudAccount { Id = metadata["accountId"].Value<string>() },
+					Instance = new CloudInstance { Id = metadata["instanceId"].Value<string>() },
+					AvailabilityZone = metadata["availabilityZone"].Value<string>(),
+					Machine = new CloudMachine { Type = metadata["instanceType"].Value<string>() },
+					Provider = Provider,
+					Region = metadata["region"].Value<string>()
+				};
+
+			}
+			catch (Exception e)
+			{
+				_logger.Warning()?.LogException(e, "Unable to get {Provider} cloud metadata", Provider);

```

Comment:
Sure about the `Warning` level? Given the default for this setting is to try all `ICloudMetadataProvider` implementations, the agent will start with 3 warnings in non-cloud and with 2 if it runs in one of the cloud providers.

I feel we should use `info` here.

We have a similar situation with the central config fetcher and I see people are very confused about those warnings as well.

Reply:
Will change to `info` - agree that `warning` seems way too strong